app/scripts/HiGlassComponent.js

```diff
@@ -3209,31 +3187,23 @@ class HiGlassComponent extends React.Component {
         this.yScales[v.uid] = scaleLinear().domain(v.initialYDomain);
       }
 
-      // Add names to all the tracks
-      let looseTracks = positionedTracksToAllTracks(v.tracks);
-
-      // give tracks their default names (e.g. 'type': 'top-axis'
-      // will get a name of 'Top Axis'
-      looseTracks = this.addUidsToTracks(looseTracks);
-      looseTracks = this.addNamesToTracks(looseTracks);
```

Comment:
Are you sure this can be removed? The comment seems to suggest that this gives tracks a default name like "Top Axis". Otherwise they may be displayed as "undefined" when a label is selected.

Reply:
Yes, this function is useless as far as I can tell. First, it overwrites any existing name and second, it adds the name to the track root object instead of the `options` property so it's anyway not used.

This used to be the function for reference:

```
const config = this.getTrackInfo(track.type);
if (config && config.name) track.name = config.name;
```

As far as I can tell `track.name` isn't used anywhere or do I overlooking something?

Reply:
Yeah, I suppose you're right. Maybe just quickly check to make sure that the Top Axis track still has a label without this function?